File: programers/for_beginner/120863.java
//다항식 더하기
import java.util.*;
class Solution {
    public String solution(String polynomial) {
        String answer;
        String[] arr= polynomial.split(" ");
        int xnum = (int)Arrays.stream(arr).filter(x->x.contains("x")&&!x.equals("x")).map(x-> x.substring(0,x.indexOf("x"))).mapToInt(Integer::parseInt).reduce(0,(a,b)->a+b)+ (int)Arrays.stream(arr).filter(x->x.equals("x")).count();
        int num = (int)Arrays.stream(arr).filter(x-> !x.contains("x") && !x.equals("+")).mapToInt(Integer::parseInt).reduce(0,(a,b)->a+b);
      if(xnum==0){
          answer= (num!=0)? Integer.toString(num):"";
      }else if(xnum==1){
          answer= (num!=0) ? "x + "+Integer.toString(num) : "x";
      }else{
          answer= (num!=0) ? Integer.toString(xnum)+"x + "+Integer.toString(num) : Integer.toString(xnum)+"x";
      }
        return answer;
    }
}

File: programers/120863.py
# 다항식 더하기
def solution(polynomial) -> str:
    ans = ""
    polynomial = polynomial.replace(" ", "")
    nums = polynomial.split("+")
    x_nums = []
    static_nums = []
    for num in nums:
        if "x" in num:
            if len(num) == 1:
                x_nums.append(1)
            else:
                x_nums.append(int(num[:-1]))
        else:
            static_nums.append(int(num))
    if n := sum(x_nums):
        if n == 1:
            ans += "x"
        else:
            ans += f"{n}x"
    if n := sum(static_nums):
        if not n:
            if not ans:
                ans += "0"
        else:
            ans = ans + f" + {n}" if ans else f"{n}"
    return ans
